src/site/topic11.rst

```diff
@@ -44,9 +44,48 @@ Enqueue
 -------
 
 
-Dequeue
--------
-
+Dequeue & First
+---------------
+
+.. code-block:: java
+    :linenos:
+
+    @Override
+    public T dequeue() {
+        if (isEmpty()) {
+            throw new NoSuchElementException("Dequeueing from an empty queue.");
+        }
+        T returnElement = front.getData();
+        front = front.getNext();
+        size--;
+        if (isEmpty()) {
+            rear = null;
+        }
+        return returnElement;
+    }
+
+    @Override
+    public T first() {
+        if (isEmpty()) {
+            throw new NoSuchElementException("First from an empty queue.");
+        }
+        return front.getData();
+    }
+
+* Like ``LinkedStack`` and ``ArrayStack``, trying to access something from the empty queue throws an exception
+
+* Notice in ``dequeue`` that, if it's not empty, we just *remove/delete from the front of a linked structure*
+    * This was the same as ``pop`` in the ``LinkedStack``
+
+* In addition to being empty, the only other edge case we need to watch out for is if the ``dequeue`` makes the queue empty
+* If this happens, we must set ``front`` to ``null``
+    * This is actually taken care of already since ``front``'s next would be ``null``, and saying ``front = front.getNext()`` makes ``front`` ``null``
+
+* We must also set ``rear`` to ``null``, otherwise it will continue to point to a node that should not be in the queue anymore
+    * Imagine ``enqueuing`` after this and messing up our counts
+    * If we ``enqueue`` in this case, we might end up saying ``rear.setNext(toEnqueue)``
```

Comment:
Comment on #110. Code does what you tell it to do, and your code says if the queue is empty when adding, update `first` and `last` to the new sole element.

Reply:
I'll add clarification on potential issue.